test/lit/passes/dae_all-features.wast

```diff
@@ -538,3 +538,115 @@
   )
  )
 )
+
+;; Arguments that read an immutable global can be optimized, as that is a
+;; constant value.
+(module
+ ;; CHECK:      (type $none_=>_none (func))
+
+ ;; CHECK:      (type $i32_=>_none (func (param i32)))
+
+ ;; CHECK:      (type $i32_i32_=>_none (func (param i32 i32)))
+
+ ;; CHECK:      (global $imm i32 (i32.const 42))
+ (global $imm i32 (i32.const 42))
```

Comment:
Super nit: How about `immut` instead of `imm` in this file? `immut`/`mut` feels more intuitive and `imm` is often used to denote immdiates elsewhere in Wasm spec, so..